source/server/config/network/mongo_proxy.h

```diff
@@ -18,6 +18,8 @@ class MongoProxyFilterConfigFactory : public NamedNetworkFilterConfigFactory {
   // NamedNetworkFilterConfigFactory
   NetworkFilterFactoryCb createFilterFactory(const Json::Object& config,
                                              FactoryContext& context) override;
+  NetworkFilterFactoryCb createFilterFactoryFromProto(const Protobuf::Message& config,
```

Comment:
You will also need to provider the factory https://github.com/envoyproxy/envoy/blob/master/include/envoy/server/filter_config.h#L173.

Reply:
Can we make sure we have tests that cover both the legacy and factory config flows also when you fix this. Thank you!

Reply:
I have not touched any existing tests other than fixing up the class names. And the existing tests use JSON config for testing. i.e. they exercise the conversion and tests pass.

Reply:
Right my point is can you please make sure there no native proto loading path for v2 config is covered, and not just the conversion path. You may need to add new tests. I think this is important as we add the rest of the filters.